[Index Management] migrate to rtl

kapral18 opened the pull request:
Closes #241537

## Summary
- Replace Index Management’s Enzyme-based Jest coverage with React Testing Library (RTL).
- Add shared RTL test infrastructure (render helpers, actions, mocks) and expand EUI RTL harnesses used across the suite.
- Standardize extracted helper-module naming under `__jest__` to `*.helpers.ts(x)` (avoids ambiguity with source-adjacent helpers).
- Harden and standardize `@kbn/test-eui-helpers` RTL harness APIs (new harnesses + more consistent method naming/null-safety) and update call sites accordingly.

## Scope
- **EUI RTL harnesses (`@kbn/test-eui-helpers`)**
  - Add/expand harnesses used heavily by Index Management tests (combo box, super select, list, pagination).
  - Add harnesses for context menus / popover panels used by various popover-driven interactions.
  - Make harness querying/selection more robust against EUI’s DOM patterns (e.g. space-separated `data-test-subj`, listbox portals).
  - Standardize harness APIs for consistency across the suite (e.g. `getElement()`, `getOptions()`, `getSelected()`, clearer method names, null-returning accessors where appropriate).
  - Remove unused harness utilities that are no longer needed by the migrated tests.

- **Index Management Jest infrastructure (`x-pack/.../index_management/__jest__/client_integration/helpers`)**
  - Introduce shared render helpers to keep individual test files focused on behavior rather than wiring.
  - Introduce small, explicit “actions” helpers to reduce repetition without building a testbed/actions DSL.
  - Add a local `@kbn/code-editor` mock for JSDOM stability.
  - Add lightweight fake-timer utilities used by the small subset of suites that intentionally opt into fake timers.

- **RTL migrations (feature areas)**
  - **Index template wizard**: migrate create/edit/clone flows and keep step interactions readable via local `*.helpers.tsx`.
  - **Index details page**: migrate main flows and supporting modals/overlays; keep waits tied to explicit UI boundaries.
  - **Home tab**: migrate indices/templates/data streams/enrich policies flows; add focused pagination coverage.
  - **Create enrich policy**: migrate the client integration flow and remove the legacy helper module.
  - **Index table component**: replace Enzyme-based component tests (and snapshots) with RTL tests and local helpers.
  - **Component templates**: migrate list/details/edit/create flows; modernize tests to TSX and consolidate helpers.
  - **Mappings editor**: migrate the client integration suite (including datatypes) and add shared fixtures where needed.

- **Server/config alignment**
  - Update Index Management server route tests and supporting mocks to match the new test setup.
  - Update `tsconfig.json`/`moon.yml` inputs so the new/relocated test files are correctly included.

## Test plan
- [ ] CI green
- [ ] Run Index Management Jest suites locally
- [ ] Sanity-check Index Management UI flows covered by the migrated tests

kapral18 committed:
chore(test-eui-helpers): expand RTL EUI harnesses

- Add RTL harnesses for combo box, list, pagination, super select
- Tighten select/table harness behavior and update exports
- Remove unused harnesses

kapral18 committed:
chore(index-management): add RTL jest test infrastructure

- Add shared render + action helpers for RTL client-integration tests
- Add local @kbn/code-editor mock and fake-timers utilities
- Simplify client-integration helper exports

kapral18 committed:
chore(index-management): migrate index template wizard tests to RTL

- Migrate template create/edit/clone client integration tests to RTL
- Extract local helpers into *.helpers.tsx under __jest__ for readability

kapral18 committed:
chore(index-management): migrate index details tests to RTL

- Migrate index details client integration tests to RTL
- Extract local helpers into *.helpers.ts(x) under __jest__ for readability

kapral18 committed:
chore(index-management): migrate home client integration tests to RTL

- Migrate Index Management home client integration tests to RTL
- Refactor local helpers to keep tests readable

kapral18 committed:
chore(index-management): migrate create enrich policy test to RTL

- Migrate create enrich policy client integration test to RTL
- Remove legacy helper module now inlined/centralized

kapral18 committed:
chore(index-management): migrate index table component tests to RTL

- Replace Enzyme-based index table component tests with RTL equivalents
- Extract local helpers to keep test cases readable

kapral18 committed:
chore(index-management): migrate component templates tests to RTL

- Migrate component templates client integration tests to RTL
- Extract local helpers into *.helpers.tsx under __jest__ for readability

kapral18 committed:
chore(index-management): migrate mappings editor tests to RTL

- Migrate mappings editor client integration tests to RTL
- Extract local helpers into *.helpers.tsx under __jest__ and add shared fixtures where needed

kapral18 committed:
chore(index-management): align server tests and tsconfig for RTL migration

- Tidy Index Management server route tests after RTL migration
- Update tsconfig/moon inputs for new test file layout
- Drop unused server test typing helper

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376780)
* Commit: 4c8ffe2c6d3367089ff50cbc84eee456c1abdc78

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/376780#019b57f1-bdb7-48b5-ad52-19e461df686f)

### History
* :yellow_heart: [Build #375921](https://buildkite.com/elastic/kibana-pull-request/builds/375921) was flaky 3da282fced7e20cac1795850e984c2bc076cf6a4
* :broken_heart: [Build #375918](https://buildkite.com/elastic/kibana-pull-request/builds/375918) failed 717753dcaed4e5995d58abd00dc03ff53e619f26
* :broken_heart: [Build #375914](https://buildkite.com/elastic/kibana-pull-request/builds/375914) failed ae5db503549cac3ec8e4abbf62ea357339f06afe
* :broken_heart: [Build #375913](https://buildkite.com/elastic/kibana-pull-request/builds/375913) failed 3300ed34e1fdd58b14ed5675b60318d26b801ccd
* :broken_heart: [Build #375912](https://buildkite.com/elastic/kibana-pull-request/builds/375912) failed af3a6f5cba9278dac014362be0b8ef44f4ed0140
* :broken_heart: [Build #375908](https://buildkite.com/elastic/kibana-pull-request/builds/375908) failed 4e345aa2f47390cccf40902042493b87482621a2

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376780","number":376780,"commit":"4c8ffe2c6d3367089ff50cbc84eee456c1abdc78"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375921","number":375921,"commit":"3da282fced7e20cac1795850e984c2bc076cf6a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375918","number":375918,"commit":"717753dcaed4e5995d58abd00dc03ff53e619f26"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375914","number":375914,"commit":"ae5db503549cac3ec8e4abbf62ea357339f06afe"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375913","number":375913,"commit":"3300ed34e1fdd58b14ed5675b60318d26b801ccd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375912","number":375912,"commit":"af3a6f5cba9278dac014362be0b8ef44f4ed0140"}],"number":376780}
buildkite-pr-comment-->

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/__jest__/client_integration/index_template_wizard/template_edit.helpers.tsx:
```diff
@@ -0,0 +1,152 @@
+    fireEvent.click(screen.getByTestId('nextButton'));
+    await screen.findByTestId('summaryTabContent');
+  },
+};
```
This looks quite similar to the `completeStep` method in `template_clone.helpers.tsx`. Maybe we can create a shared helper to avoid redundant code. Can be done as a follow up

SoniaSanzV commented on src/platform/packages/private/kbn-test-eui-helpers/src/rtl/eui_pagination_test_harness.tsx:
```diff
@@ -0,0 +1,41 @@
+  }
+
+  public get buttons(): HTMLButtonElement[] {
+    return Array.from(this.#root.querySelectorAll<HTMLButtonElement>('button.euiPaginationButton'));
```
Can we avoid using css selectors?

kapral18 replied:
I believe this was a move from existing code, but yepp I will tweak, thanks

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/__jest__/components/index_table.helpers.tsx:
```diff
@@ -0,0 +1,242 @@
+  const manageButton = await screen.findByTestId('indexActionsContextMenuButton');
+  fireEvent.click(manageButton);
+
+  const menu = await screen.findByTestId('indexContextMenu');
```
this is done a lot of times! We could use a helper function

kapral18 replied:
done

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/public/application/components/component_templates/__jest__/client_integration/component_template_edit.test.tsx:
```diff
@@ -5,72 +5,39 @@
 
-  const { httpSetup, httpRequestsMockHelpers } = setupEnvironment();
-  jest.spyOn(breadcrumbService, 'setBreadcrumbs');
+  afterAll(() => {});
```
this is empty, should be removed

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/public/application/components/mappings_editor/__jest__/client_integration/mappings_editor.test.tsx:
```diff
@@ -44,19 +65,167 @@ describe('Mappings editor: core', () => {
-  beforeAll(() => {
-    jest.useFakeTimers({ legacyFakeTimers: true });
-  });
+  // RTL helper functions
```
Why is this information useful?

kapral18 replied:
sorry didn't notice how this file slipped through. thanks for reminding. I just [pushed](https://github.com/elastic/kibana/pull/244909/changes/7477f696aaf3eac388dbc1996b18deaa98fdf8ac#diff-87e6d52c8915289b87d02aee38975f85d9cf1380507af9aade279a69db4e5985). rtl redundant comments no longer show up.

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/public/application/components/mappings_editor/__jest__/client_integration/mappings_editor.test.tsx:
```diff
@@ -344,290 +529,308 @@ describe('Mappings editor: core', () => {
+      test('props.value => should prepopulate the editor data', async () => {
+        /**
+         * Mapped fields
+         */
```
```suggestion
// Mapped fields
```

kapral18 replied:
done

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/public/application/components/mappings_editor/__jest__/client_integration/mappings_editor.test.tsx:
```diff
@@ -344,290 +529,308 @@ describe('Mappings editor: core', () => {
+
+        /**
+         * Dynamic templates
+         */
```
```suggestion
// Dynamic templates
```

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/public/application/components/mappings_editor/__jest__/client_integration/mappings_editor.test.tsx:
```diff
@@ -344,290 +529,308 @@ describe('Mappings editor: core', () => {
-      component.update();
+        /**
+         * Mapped fields
+         */
```
```suggestion
// Mapped fields
```

SoniaSanzV reviewed (approved):
Great work @kapral18!! I left a few comments but if there is a rush to merge the pr, they can be addressed as a follow-up!

kapral18 commented on x-pack/platform/plugins/shared/index_management/__jest__/client_integration/home/enrich_policies.test.tsx:
```diff
@@ -43,17 +40,15 @@ describe('Enrich policies tab', () => {
+      expect(await screen.findByTestId('sectionLoading')).toBeInTheDocument();
 
-      expect(exists('sectionLoading')).toBe(true);
+      await renderPromise;
```
this was artifact from earlier fakeTimers based implementation, where the only way to catch sectionLoading state was by freezing the promise, since I moved away from that this was a leftover. thanks

SoniaSanzV commented on x-pack/platform/plugins/shared/index_management/__jest__/client_integration/index_template_wizard/template_create.test.tsx:
```diff
@@ -673,24 +533,56 @@ describe('<TemplateCreate />', () => {
   });
 
-  describe.skip('DSL', () => {
+  describe('DSL', () => {
```
oh, make sense. I was thinking in Data Lifecycle Management

kapral18 replied:
addressed

kapral18 committed:
chore: address comments

kapral18 commented on x-pack/platform/plugins/shared/index_management/__jest__/client_integration/index_template_wizard/template_clone.test.tsx:
```diff
@@ -5,113 +5,90 @@
-    await act(async () => {
-      testBed = await setup(httpSetup);
+  // RTL test - migrated
+  describe('page title (RTL)', () => {
```
done

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376988)
* Commit: 9c718f6f58d853bbc9f35d45caa194e612f70805

### Failed CI Steps
* [FTR Configs #79](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6ac9-f12f-49ea-a1f4-581ab19a3d93)
* [FTR Configs #87](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6ac9-f138-4893-8a7a-60518859f2ca)
* [FTR Configs #87](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6aef-19c0-4130-a660-d7bbbfe5e401)
* [Scout: [ observability / apm ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6acd-dead-480a-aa54-780f28f15735)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6acd-dead-480a-aa54-780f28f15735) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376988/jobs/019b6acd-dead-480a-aa54-780f28f15735/artifacts/019b6adf-069d-495f-815f-f07def2a06ca) Scout: [ observability / apm ] plugin / Can create, trigger and view an 'Error count' alert from service inventory
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6ac9-f12f-49ea-a1f4-581ab19a3d93) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376988/jobs/019b6ac9-f12f-49ea-a1f4-581ab19a3d93/artifacts/019b6aef-4eef-48bb-9d81-021cf72778aa) FTR Configs #<span></span>79 / Cloud Security Posture Security Network Page - Graph visualization ECS fields only expanded flyout - entity enrichment for multiple actors and targets
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6acd-dead-480a-aa54-780f28f15735) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376988/jobs/019b6acd-dead-480a-aa54-780f28f15735/artifacts/019b6adf-069d-464e-87d9-b21e18845531) Scout: [ observability / apm ] plugin / serverless-oblt - Alerts - Can create, trigger and view an 'Error count' alert from service inventory
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6ac9-f138-4893-8a7a-60518859f2ca) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376988/jobs/019b6ac9-f138-4893-8a7a-60518859f2ca/artifacts/019b6aef-024e-4412-a127-9cbc214c29cb) FTR Configs #<span></span>87 / task_manager scheduling and running tasks should schedule a task with rrule with fixed time and dtstart
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376988#019b6aef-19c0-4130-a660-d7bbbfe5e401) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376988/jobs/019b6aef-19c0-4130-a660-d7bbbfe5e401/artifacts/019b6af9-4ba8-4183-8677-3e0251759dc1) FTR Configs #<span></span>87 / task_manager scheduling and running tasks should schedule a task with rrule with fixed time and dtstart

### History
* :green_heart: [Build #376957](https://buildkite.com/elastic/kibana-pull-request/builds/376957) succeeded 27155fa8f341f0ca9a9b0b166dde370524dc2b52
* :broken_heart: [Build #376780](https://buildkite.com/elastic/kibana-pull-request/builds/376780) failed 4c8ffe2c6d3367089ff50cbc84eee456c1abdc78
* :yellow_heart: [Build #375921](https://buildkite.com/elastic/kibana-pull-request/builds/375921) was flaky 3da282fced7e20cac1795850e984c2bc076cf6a4
* :broken_heart: [Build #375918](https://buildkite.com/elastic/kibana-pull-request/builds/375918) failed 717753dcaed4e5995d58abd00dc03ff53e619f26
* :broken_heart: [Build #375914](https://buildkite.com/elastic/kibana-pull-request/builds/375914) failed ae5db503549cac3ec8e4abbf62ea357339f06afe
* :broken_heart: [Build #375913](https://buildkite.com/elastic/kibana-pull-request/builds/375913) failed 3300ed34e1fdd58b14ed5675b60318d26b801ccd

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376988","number":376988,"commit":"9c718f6f58d853bbc9f35d45caa194e612f70805"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376957","number":376957,"commit":"27155fa8f341f0ca9a9b0b166dde370524dc2b52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376780","number":376780,"commit":"4c8ffe2c6d3367089ff50cbc84eee456c1abdc78"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375921","number":375921,"commit":"3da282fced7e20cac1795850e984c2bc076cf6a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375918","number":375918,"commit":"717753dcaed4e5995d58abd00dc03ff53e619f26"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375914","number":375914,"commit":"ae5db503549cac3ec8e4abbf62ea357339f06afe"}],"number":376988}
buildkite-pr-comment-->

kapral18 committed:
chore(kbn-test-eui-helpers): improve test harness API consistency

- Add `EuiContextMenuTestHarness` for context menu interactions in popovers
- Add `EuiPopoverPanelTestHarness` for discovering popover panels and their menus
- Standardize element accessor naming: `getElement()` replaces `self` getter
- Standardize accessor naming: `getOptions()`, `getSelected()` replace `options`, `selected` getters
- Convert throwing getters to null-returning methods where appropriate
- Rename methods for clarity: `selectOption` → `select`, `clickItemAction` → `clickAction`
- Convert static methods to instance methods with optional root context
- Simplify `EuiTableTestHarness` row access to return null instead of throwing
- Update all test files to use new harness APIs

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377052)
* Commit: 08df5a1dcb8e69464dc751de5a7935c61ff595e0

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/377052#019b6cec-1c1d-48cc-bedc-db269ef17993)

### History
* :broken_heart: [Build #376988](https://buildkite.com/elastic/kibana-pull-request/builds/376988) failed 9c718f6f58d853bbc9f35d45caa194e612f70805
* :green_heart: [Build #376957](https://buildkite.com/elastic/kibana-pull-request/builds/376957) succeeded 27155fa8f341f0ca9a9b0b166dde370524dc2b52
* :broken_heart: [Build #376780](https://buildkite.com/elastic/kibana-pull-request/builds/376780) failed 4c8ffe2c6d3367089ff50cbc84eee456c1abdc78
* :yellow_heart: [Build #375921](https://buildkite.com/elastic/kibana-pull-request/builds/375921) was flaky 3da282fced7e20cac1795850e984c2bc076cf6a4
* :broken_heart: [Build #375918](https://buildkite.com/elastic/kibana-pull-request/builds/375918) failed 717753dcaed4e5995d58abd00dc03ff53e619f26
* :broken_heart: [Build #375914](https://buildkite.com/elastic/kibana-pull-request/builds/375914) failed ae5db503549cac3ec8e4abbf62ea357339f06afe

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377052","number":377052,"commit":"08df5a1dcb8e69464dc751de5a7935c61ff595e0"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376988","number":376988,"commit":"9c718f6f58d853bbc9f35d45caa194e612f70805"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376957","number":376957,"commit":"27155fa8f341f0ca9a9b0b166dde370524dc2b52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376780","number":376780,"commit":"4c8ffe2c6d3367089ff50cbc84eee456c1abdc78"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375921","number":375921,"commit":"3da282fced7e20cac1795850e984c2bc076cf6a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375918","number":375918,"commit":"717753dcaed4e5995d58abd00dc03ff53e619f26"}],"number":377052}
buildkite-pr-comment-->

kapral18 committed:
chore: address comment leftover and ci type regressions

walterra reviewed (approved):
Code review for vis code owners LGTM. 👍

kapral18 committed:
chore: fix ccr ci regressions

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377094)
* Commit: dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `@kbn/test-eui-helpers` | 32 | 87 | +55 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 732.5KB | 732.5KB | +12.0B |

#### Count of Enzyme imports
> Enzyme is no longer supported, and we should switch to @testing-library/react instead.


| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 3 | 0 | -3 |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `@kbn/test-eui-helpers` | 0 | 1 | +1 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `@kbn/test-eui-helpers` | 63 | 138 | +75 |

#### ESLint disabled in files
| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 2 | 1 | -1 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 15 | 12 | -3 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 65 | 10 | -55 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/16a5dff4252276c93daba304c533439b5383b4ad) | [after](https://github.com/elastic/kibana/commit/dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 17 | 13 | -4 |


</details>


### History
* :broken_heart: [Build #377079](https://buildkite.com/elastic/kibana-pull-request/builds/377079) failed 7477f696aaf3eac388dbc1996b18deaa98fdf8ac
* :broken_heart: [Build #377052](https://buildkite.com/elastic/kibana-pull-request/builds/377052) failed 08df5a1dcb8e69464dc751de5a7935c61ff595e0
* :broken_heart: [Build #376988](https://buildkite.com/elastic/kibana-pull-request/builds/376988) failed 9c718f6f58d853bbc9f35d45caa194e612f70805
* :green_heart: [Build #376957](https://buildkite.com/elastic/kibana-pull-request/builds/376957) succeeded 27155fa8f341f0ca9a9b0b166dde370524dc2b52
* :broken_heart: [Build #376780](https://buildkite.com/elastic/kibana-pull-request/builds/376780) failed 4c8ffe2c6d3367089ff50cbc84eee456c1abdc78
* :yellow_heart: [Build #375921](https://buildkite.com/elastic/kibana-pull-request/builds/375921) was flaky 3da282fced7e20cac1795850e984c2bc076cf6a4

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377094","number":377094,"commit":"dabe135ba8192e85bd2ef8ee5f0c92704f10ee2b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377079","number":377079,"commit":"7477f696aaf3eac388dbc1996b18deaa98fdf8ac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377052","number":377052,"commit":"08df5a1dcb8e69464dc751de5a7935c61ff595e0"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376988","number":376988,"commit":"9c718f6f58d853bbc9f35d45caa194e612f70805"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376957","number":376957,"commit":"27155fa8f341f0ca9a9b0b166dde370524dc2b52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376780","number":376780,"commit":"4c8ffe2c6d3367089ff50cbc84eee456c1abdc78"}],"number":377094}
buildkite-pr-comment-->
